src/Components/Artwork/Details/PrimaryLabelLine.tsx

```diff
@@ -1,27 +1,44 @@
 import { Text } from "@artsy/palette"
 import { useArtworkGridContext } from "Components/ArtworkGrid/ArtworkGridContext"
 import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
+import { useFulfillmentOptionsForArtwork } from "Utils/Hooks/useFulfillmentOptions"
 import type { PrimaryLabelLineQuery } from "__generated__/PrimaryLabelLineQuery.graphql"
 import type { PrimaryLabelLine_artwork$data } from "__generated__/PrimaryLabelLine_artwork.graphql"
+import type { PrimaryLabelLine_me$data } from "__generated__/PrimaryLabelLine_me.graphql"
 import { type FC, useEffect } from "react"
 import { createFragmentContainer, graphql } from "react-relay"
 
 interface PrimaryLabelLineProps {
   label: string | null | undefined
   artwork?: PrimaryLabelLine_artwork$data
+  me?: PrimaryLabelLine_me$data
+}
+
+const doesShipFree = (
+  artwork: PrimaryLabelLine_artwork$data,
+  me?: PrimaryLabelLine_me$data,
+): boolean => {
+  const result = useFulfillmentOptionsForArtwork(artwork, me)
```

Comment:
This doesn't seem right as this manner of using a hook in a function breaks the rules of hooks

Reply:
oh good point. I was just trying to get it working since if implemented we would probably move at least some of this logic upstream to MP/Gravity